File: 이진검색(이진탐색)/leetcode1539(Easy)-Kth Missing Positive Number.py
# O(logN)의 이분탐색 : 56ms(55.94%), 14.1MB(44.77%)
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        left, right = 0, len(arr)

        while left < right:
            mid = (left + right) // 2

            if arr[mid] <= k + mid:
                left = mid + 1
            else:
                right = mid
        
        return left + k

File: 이진검색(이진탐색)/leetcode1539(Easy)-Kth Missing Positive Number.js
// 이분탐색 (logN) : 54ms(94%), 42MB(67%)
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function (arr, k) {
  let left = 0
  let right = arr.length

  while (left < right) {
    const mid = Math.floor((left + right) / 2)

    if (arr[mid] <= k + mid) {
      left = mid + 1
    } else {
      right = mid
    }
  }

  return left + k
}
